openshift-knative-operator/pkg/monitoring/monitoring.go

```diff
@@ -109,3 +122,87 @@ func InjectNamespaceWithSubject(resourceNamespace string, subjectNamespace strin
 		return nil
 	}
 }
+
+func LoadServingMonitoringPlatformManifests(ns string) ([]mf.Manifest, error) {
+	rbacPath := os.Getenv(servingSMRbacManifestPath)
+	if rbacPath == "" {
+		return nil, fmt.Errorf("failed to get the Serving sm rbac manifest path")
+	}
+	rbacManifest, err := mf.NewManifest(rbacPath)
+	if err != nil {
+		return nil, err
+	}
+	for c := range servingComponents {
+		smManifest, err := constructServiceMonitorResourceManifests(c, ns)
+		if err != nil {
+			return nil, err
+		}
+		if smManifest != nil {
+			rbacManifest = rbacManifest.Append(*smManifest)
+		}
+	}
+	return []mf.Manifest{rbacManifest}, nil
+}
+
+func constructServiceMonitorResourceManifests(component string, ns string) (*mf.Manifest, error) {
+	var smU = &unstructured.Unstructured{}
+	var svU = &unstructured.Unstructured{}
+	sm := createServiceMonitor(component, ns)
+	if err := scheme.Scheme.Convert(&sm, smU, nil); err != nil {
+		return nil, err
+	}
+	sms := createServiceMonitorService(component)
+	if err := scheme.Scheme.Convert(&sms, svU, nil); err != nil {
+		return nil, err
+	}
+	smManifest, err := mf.ManifestFrom(mf.Slice([]unstructured.Unstructured{*smU, *svU}))
+	if err != nil {
+		return nil, err
+	}
+	return &smManifest, nil
+}
+
+func createServiceMonitor(component string, ns string) monitoringv1.ServiceMonitor {
+	return monitoringv1.ServiceMonitor{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-sm", component),
+		},
+		Spec: monitoringv1.ServiceMonitorSpec{
+			Endpoints: []monitoringv1.Endpoint{
+				{
```

Comment:
nit: make it drop a level of indentation I believe.

```suggestion
			Endpoints: []monitoringv1.Endpoint{{
```